src/samples/ExtendedSlashScreen.Uno.Samples/ExtendedSlashScreen.Uno.Samples.Shared/App.xaml.cs

```diff
@@ -1,112 +1,111 @@
-using System.IO;
-using System.Linq;
-using System.Runtime.InteropServices.WindowsRuntime;
 using Windows.ApplicationModel;
 using Windows.ApplicationModel.Activation;
-using Windows.Foundation;
-using Windows.Foundation.Collections;
+#if WINUI
+using Microsoft.UI.Xaml;
+using LaunchActivatedEventArgs = Microsoft.UI.Xaml.LaunchActivatedEventArgs;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+#else
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Controls;
-using Windows.UI.Xaml.Controls.Primitives;
-using Windows.UI.Xaml.Data;
-using Windows.UI.Xaml.Input;
-using Windows.UI.Xaml.Media;
 using Windows.UI.Xaml.Navigation;
+#endif
 
 namespace ExtendedSlashScreen.Uno.Samples
 {
-    /// <summary>
-    /// Provides application-specific behavior to supplement the default Application class.
-    /// </summary>
-    sealed partial class App : Application
-    {
-        /// <summary>
-        /// Initializes the singleton application object.  This is the first line of authored code
-        /// executed, and as such is the logical equivalent of main() or WinMain().
-        /// </summary>
-        public App()
-        {
-            ConfigureFilters(global::Uno.Extensions.LogExtensionPoint.AmbientLoggerFactory);
-
-            this.InitializeComponent();
-            this.Suspending += OnSuspending;
-        }
-
-		public Shell Shell { get; private set; }
-
-		public Window CurrentWindow => Windows.UI.Xaml.Window.Current;
-
-		public Activity ShellActivity { get; } = new Activity(nameof(Shell));
+	/// <summary>
+	/// Provides application-specific behavior to supplement the default Application class.
+	/// </summary>
+	public sealed partial class App : Application
+	{
+		/// <summary>
+		/// Initializes the singleton application object.  This is the first line of authored code
+		/// executed, and as such is the logical equivalent of main() or WinMain().
+		/// </summary>
+		public App()
+		{
+			ConfigureFilters(global::Uno.Extensions.LogExtensionPoint.AmbientLoggerFactory);
+
+			this.InitializeComponent();
+#if !WINUI
+			this.Suspending += OnSuspending;
+#endif
+		}
 
 		/// <summary>
 		/// Invoked when the application is launched normally by the end user.  Other entry points
 		/// will be used such as when the application is launched to open a specific file.
 		/// </summary>
 		/// <param name="e">Details about the launch request and process.</param>
 		protected override void OnLaunched(LaunchActivatedEventArgs e)
-        {
+		{
 #if DEBUG
 			if (System.Diagnostics.Debugger.IsAttached)
 			{
 				// this.DebugSettings.EnableFrameRateCounter = true;
 			}
-# endif
-			Shell = CurrentWindow.Content as Shell;
-
-			var isFirstLaunch = Shell == null;
-
-			if (isFirstLaunch)
+#endif
+
+#if WINDOWS
+			var window = new Window();
+			window.Activate();
+#elif __IOS__ && HAS_UNO_WINUI
```

Comment:
Instead of HAS_UNO_WINUI, we use the constant WINUI that is defined in the WINUI.csproj